installer/x86_64/install.sh

```diff
@@ -481,13 +481,18 @@ fi
 # Decompress the file for the file system directly to the partition
 unzip -o $ONIE_INSTALLER_PAYLOAD -x "$FILESYSTEM_DOCKERFS" -d $demo_mnt/$image_dir
 
-if [ "$install_env" = "onie" ]; then
-    TAR_EXTRA_OPTION="--numeric-owner"
+# when disk is small, keep dockerfs.tar.gz in disk, expand it into ramfs during initrd
+if [ x"$DOCKER_IN_RAM" = x"y" ]; then
+    unzip -o $ONIE_INSTALLER_PAYLOAD "$FILESYSTEM_DOCKERFS" -d $demo_mnt/$image_dir
 else
-    TAR_EXTRA_OPTION="--numeric-owner --warning=no-timestamp"
+    if [ "$install_env" = "onie" ]; then
+        TAR_EXTRA_OPTION="--numeric-owner"
+    else
+        TAR_EXTRA_OPTION="--numeric-owner --warning=no-timestamp"
+    fi
+    mkdir -p $demo_mnt/$image_dir/$DOCKERFS_DIR
+    unzip -op $ONIE_INSTALLER_PAYLOAD "$FILESYSTEM_DOCKERFS" | tar xz $TAR_EXTRA_OPTION -f - -C $demo_mnt/$image_dir/$DOCKERFS_DIR
 fi
-mkdir -p $demo_mnt/$image_dir/$DOCKERFS_DIR
-unzip -op $ONIE_INSTALLER_PAYLOAD "$FILESYSTEM_DOCKERFS" | tar xz $TAR_EXTRA_OPTION -f - -C $demo_mnt/$image_dir/$DOCKERFS_DIR
 
```

Comment:
```
# when disk is small, keep dockerfs.tar.gz in disk, expand it into ramfs during initrd
if [ x"$DOCKER_IN_RAM" = x"y" ]; then
    unzip -o $ONIE_INSTALLER_PAYLOAD -d $demo_mnt/$image_dir
else
    # Decompress the file for the file system directly to the partition, excluding dockerfs.tar.gz
    unzip -o $ONIE_INSTALLER_PAYLOAD -x "$FILESYSTEM_DOCKERFS" -d $demo_mnt/$image_dir

    if [ "$install_env" = "onie" ]; then
        TAR_EXTRA_OPTION="--numeric-owner"
    else
        TAR_EXTRA_OPTION="--numeric-owner --warning=no-timestamp"
    fi
    mkdir -p $demo_mnt/$image_dir/$DOCKERFS_DIR
    unzip -op $ONIE_INSTALLER_PAYLOAD "$FILESYSTEM_DOCKERFS" | tar xz $TAR_EXTRA_OPTION -f - -C $demo_mnt/$image_dir/$DOCKERFS_DIR
fi
```

Reply:
not sure why you want to change this.

unzip -o $ONIE_INSTALLER_PAYLOAD "$FILESYSTEM_DOCKERFS" -d $demo_mnt/$image_dir

Reply:
The reason is you can use one unzip command instead of two for the first branch. To clarify, my code is trying to replace the block starting from the outside unzip line above if-else.

---
In reply to: [310447996](https://github.com/Azure/sonic-buildimage/pull/3279#discussion_r310447996) [](ancestors = 310447996)